infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -552,6 +589,22 @@ private void handleAllocation(CorfuPayloadMsg<TokenRequest> msg, ChannelHandlerC
                     return newTail - 1;
                 }
             });
+
+            // step 3. add allocated addresses to each stream's address map (to keep track of all updates to this stream)
+            streamsAddressSpaceMap.compute(id, (k, v) -> {
```

Comment:
I wouldn't like to use `compute()` method, it is overcomplicated, it comes very hard to change/understand/maintain code in the future. Can you please refactor this?

Reply:
I'm trying to be consistent also with what is already part of this code. If you see the step before it is done in the same way. A mix of too many styles for achieving the same things is a bit confusing in my opinion and not consistent in the code.

Reply:
there are no styles, there is only `cognitive complexity`, if the complexity is low then code quality is high and vice versa.
Imagine this situation: 
1. you have 100 lines of bad code, you follow "the style" and write 20 lines of bad code (because you follow the style) and change 10 lines of bad code, but since you follow "style" you don't improve things, and so on and so on, after some number of iterations you end up with very low-quality code.

2. you have 100 lines of bad code, you write you 20 lines of good code and change 10 lines of bad code, now you have "2" styles, next time you change 30 lines of old code and replace it by better quality code and so on and so on. Finally, all your code has good quality.  It's called refactoring, you continuously improve your code base.

According to your logic, there is no way to improve anything you just have to follow the style. But in reality, there is no any "style", any app a huge number of approaches written by different people and following to the code that already written is a pretty bad idea. Only thing that you can do is revrite everything with the new style which is impossible.

Reply:
https://resources.collab.net/agile-101/code-refactoring